python/ray/autoscaler/updater.py

```diff
@@ -312,14 +353,15 @@ def __init__(self,
                  ray_start_commands,
                  runtime_hash,
                  process_runner=subprocess,
-                 use_internal_ip=False):
+                 use_internal_ip=False,
+                 docker_config=None):
```

Comment:
Should we just pass the entire config in?
Currently `provider_config`, `auth_config`, `cluster_name`, and `file_mounts` (and now `docker_config`) are all parts

Reply:
No, let's pass in the docker_config like what you're doing now

Reply:
Okay--is there a specific reason for this?

Reply:
Well, it helps isolate what is needed and what is not needed. For example, initialization_commands is also part of the config, but there are times when we use the NodeUpdater where we want to run it and when we don't want to.